### PR TITLE
MZC-1205: Trend lines for correlation dashboard, mz-bokeh-package update

### PR DESCRIPTION
**Pull request by ofir-frd**

### CODE REVIEW

**Comment by ofir-frd** [comment]
This PR is attached to a parallel PR from bokeh_apps
https://github.com/materialscloud/bokeh_apps/pull/158

File: mz_bokeh_package/components/plot_settings.py [hunk 1]
```diff
@@ -357,12 +362,20 @@ def _point_shape(self, value: str):
...
+        excluded_renderers_list=[]
+        for renderer in self._plot.renderers:
+            # Deny excluded list renderers removal , such as trend line.
+            if renderer.name in self._excluded_renderers:
+                excluded_renderers_list.append(renderer)
+        self._plot.renderers = self._plot.renderers[(len(self._plot.renderers)-len(glyph_kwargs)):] + \
+                               excluded_renderers_list
+
+        # Update legend: scatter point renderers are not added to the legend
+        number_of_legend_items = len(self._plot.renderers) - len(self._plot.legend.items)
```
Comment (captKuk) [inline mz_bokeh_package/components/plot_settings.py:374]: This looks very strange to me. You first remove the renderers of the trend line from the list that `_get_glyph_kwargs` produces, and then add them again. Maybe I need to read it a bit more carefully, but so far, this doesn't make much sense to me...

File: mz_bokeh_package/components/plot_settings.py [hunk 2]
```diff
@@ -759,7 +781,7 @@ def _set_colors_by_field(self, colors: List[str]):
             colors (List[str]): A list of colors that were chosen by the user in the settings modal.
         """  # noqa: E501
 
-        renderer = self._plot.renderers[0]
+        renderer = self._plot.select(name="actualData")
```
Comment (captKuk) [inline mz_bokeh_package/components/plot_settings.py:784]: It's the same as above, and with added complication that the apps already have to implement the correct name to even start working. There is no good reason to request the default to have a special name. This makes it really complicated to use that class.

File: mz_bokeh_package/components/plot_settings.py [hunk 3]
```diff
@@ -180,6 +180,11 @@ def __init__(
         self._included_settings = included_settings or BASE_SETTINGS
         self._plot_tool_description = "Plot Settings"
 
+        # List of renderers to exclude (currently, only trend_line):
+        self._excluded_renderers = ["trend_line",
+                                    "upper_prediction_limit_intercept",
+                                    "lower_prediction_limit_intercept"]
```
Comment (captKuk) [inline mz_bokeh_package/components/plot_settings.py:186]: As mentioned in the other PR, this feels like a bad idea. The purpose of this package is to provide a general structure that is accessible to all apps. Hard-coding a very specific application into it, has many risks, just to name two: Either you will keep adding stuff to this list and bloat it, or you'll use these names in other apps to get the desired functionality, but with a different purpose. Then, communicating between package by these secretive codes has all risks of typos, and results in a lot of problems in identifying errors.

**Review by captKuk — COMMENTED** [review]
Hey @ofir-frd, 

The main technical criticism of this PR are the hard-coded renderer-names. You could build it such that the plot setting exposes a method that lets the user fill in the values in the list of `excluded_renderers`, and not use the `actualData` name-tag at all. 

However, I don't think this is a good way to deal with this problem. The conceptual criticism is that it doesn't seem to be necessary to implement a solution for a specific problem in a general class.

My first attempt would be to try to live without touching this package at all. I'm not even sure I understand what you want to solve. Is it that you want to have the trend line appear in its own legend? You could implement a legend for the trend line in the `PlotWidget` proposed in the other PR. 
The second alternative would be to work with a class inheriting from these `PlotSettings`, but only if the first approach really doesn't work.

I'm quite certain that one of these approaches should work, and if you implement them somewhat cleanly in the `PlotWidget`, you also have a general solution at hand for future migrations of the `TrendLine`.

**Pull request closed without merging**